tests/parser/syntax/test_minmax.py

```diff
@@ -1,17 +1,27 @@
 import pytest
 
-from vyper.exceptions import InvalidType
+from vyper.exceptions import InvalidType, TypeMismatch
 
 fail_list = [
-    """
+    (
+        """
 @external
 def foo():
     y: int128 = min(7, 0x1234567890123456789012345678901234567890)
-    """
+    """,
+        InvalidType,
+    ),
+    (
+        """
+@external
+def foo():
+    a: int256 = min(-1, max_value(int256) + 1)
```

Comment:
this fails for a different reason, no?

Reply:
It fails on the same branch: https://github.com/vyperlang/vyper/blob/efb00329ba6704187b0ac4255cb353c5df1b950c/vyper/builtins/functions.py#L2019-L2020

Although `max_value(int256) + 1` is out of bounds, it is currently not type-checked before folding so it is not flagged as an error. Therefore, this slightly modified example would also compile:
```
@external
def foo():
    a: int248 = min(1, max_value(int248) + 1)
```

Reply:
the example is a bit confusing tbh because the operands are not really valid to begin with. and i'm realizing the fix itself does not block examples which are in the int255 range but do not typecheck, like:
```vyper
min( min_value(int16)/*type: int16*/, max_value(int8)/*type: int8*/)
```
can we use `get_common_types` here like other builtins which deal with numeric literals? i think that can fix the typechecking issue.

Reply:
> the example is a bit confusing tbh because the operands are not really valid to begin with.

I have moved this test to #3201 instead. I think that PR is more relevant for addressing these cases.